src/scene/scene.cpp

```diff
@@ -25,6 +29,12 @@ Hit Scene::getRayIntersection(const Ray &ray) {
       hit = curHit;
     }
   }
-
+  glm::vec3 color{0, 0, 0};
+  for (const Light &light : mLightList) {
+    color += light.GetColor(mTriangleList[hit.getTriId()]);
+  }
+  if (mLightList.size() > 0) {
+    hit.setColor(color);
+  }
```

Comment:
I think this code should be moved to the color ray function. It won't make a difference now because we are just shooting one ray per pixel (with no reflections or refractions) but in the future one pixel's color will be affected by multiple ray's/hit's color so we need the lighting equation/code to be encapsulated in the colorRay function.

Reply:
Issue with that is I need mTriangleList which is not exposed out of the scene. I'm not sure I understand the concern here. Are we expecting to have more than one scene?
```
glm::vec3 PathTracer::colorRay(const Ray &ray) {
  Hit hit = mScene->getRayIntersection(ray);
  return hit.color();
}
```
colorRay as defined above just seems like a wrapper function?

Reply:
colorRay is only a wrapper function for now it will do more once we implement more features. The concern is that rays will spawn more rays and to properly color a ray you need know the color of the rays it spawned. This is how reflection works in path tracers. So doing the lighting at the end will not be sufficient for scenes with reflection and refraction. I think there are 2 solutions. Either change Scene to expose the triangle list (with like getTri(int index) using hit's mTriId). Or add a mNormal property to Hit so it can store the normal of the triangle. Then you wouldn't need to pass the triangle into Light's getColor just the normal vec.